synapse/storage/search.py

```diff
@@ -145,25 +145,30 @@ def reindex_search_txn(txn):
         defer.returnValue(result)
 
     @defer.inlineCallbacks
-    def _background_reindex_gist_search(self, progress, batch_size):
+    def _background_reindex_gin_search(self, progress, batch_size):
+        '''This handles old synapses which used GIST indexes; converting them
+        back to be GIN as per the actual schema.  Otherwise it crashes out
+        as a NOOP
```

Comment:
How does the background reindexer handle things that crash?

Reply:
ah, but a66f489678dc05fa89e6849405c37a9a390e62fc was an attempt to fix this.